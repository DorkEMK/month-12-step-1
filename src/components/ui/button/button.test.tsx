import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent, screen } from "@testing-library/react";
import { Button } from "./button";

const testData = {
  text: "test",
};

describe("Button component", () => {
  it("renders default", () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("renders with text", () => {
    const button = renderer.create(<Button text={testData.text} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("renders when disabled", () => {
    const button = renderer.create(<Button disabled />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("renders when loading", () => {
    const button = renderer.create(<Button isLoader />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it("fires callback on click", () => {
    const callback = jest.fn();
    render(<Button text={testData.text} onClick={callback}/>);
    const button = screen.getByText(testData.text);
    fireEvent.click(button);
    expect(callback).toBeCalled();
  });
});
