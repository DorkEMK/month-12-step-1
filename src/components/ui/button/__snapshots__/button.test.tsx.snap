// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders default 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component renders when disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component renders when loading 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component renders with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;
